FlashCards.AhmadJer99/FlashCards/Data/StudyDBController.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Dapper;
+using FlashCards.Models;
+
+namespace FlashCards.Data;
+
+internal class StudyDBController : DataBaseController<StudySession>
+{
+    public StudyDBController()
+    {
+        InitDataBase();
+    }
+    public override void DeleteRow(int _id) // Not needed
```

Comment:
🟠 **Redundant Code**

💡 If you have taken the time to comment you code not needed, why didn't you just remove it? In the future you might come back to this and be typing in another file, see DeleteRow exists and use it. Then bang! NotImplementedException says hello. Minor, I know, but will definitely get annoying in the future and also comes across as unprofessional.

Reply:
But the problem here is that I used an abstract class , and I can't not implement abstract methods !  what should I do in this situation ?

Reply:
Throwing `NotImplementedException` for methods like `UpdateSomething` and `DeleteSomething` suggests that the current design is not a good fit.

This can violate the Liskov Substitution Principle (the L in SOLID).

Instead, what you could have done is refactor into smaller interfaces, such as `IReadable`, `IWritable`, or even more granular `IUpdatable`, `IDeletable`.
Or you could have used Repository pattern, which is a few project down the road if I recall, but this pattern would allow you to only implement the functionality your class actually uses.

Reply:
ooh ok, i wasn't sure at the time what i can do about it , thanks for the explanation , and btw I haven't read about the SOLID yet.

Reply:
That's perfectly fine it is not a requirement, I was just explaining why it is an issue 😄 .

Reply:
Well at least if i dont fix it now , I know what to avoid next time.